common/src/test/java/com/sequenceiq/cloudbreak/common/service/LockServiceTest.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.cloudbreak.common.service;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+@ExtendWith(MockitoExtension.class)
```

Comment:
Please add a test where the runnable fails and verify that the unlock method will be called.

Reply:
Some higher level test with real postgres would be usefull as well.

Reply:
I added the mentioned unit test. But the real postgres test would be very hard to implement :S we have tests for integration testing with DB but it is based on H2.

Reply:
It is not hard, we already have example which uses postgres through test containers.
https://github.com/hortonworks/cloudbreak/blob/master/flow/src/test/java/com/sequenceiq/flow/component/FlowComponentTest.java

You can create a new component test in the required module and add it to the component test jenkins job.